app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+		},
+	)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), resR, "empty staking rewards")
+	assert.NotEmpty(s.T(), resR.Rewards)
+
+	rewardAmount := resR.Rewards[0].Reward.AmountOf(app.BondDenom).RoundInt().Int64()
+	assert.Greater(s.T(), rewardAmount, int64(0), "rewards must be more than zero")
+
+	balancesBefore, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+
+	// min is used as in the middle of the operation more tokens might be vested (unlocked)
+	minExpectedBalance := balancesBefore.AmountOf(app.BondDenom).Int64() + rewardAmount
+
+	validators, err := GetValidators(cctx.GRPCClient)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), validators, "empty validators set")
+
+	msg := distributiontypes.NewMsgWithdrawDelegatorReward(
+		getAddress(accName, cctx.Keyring),
+		validators[0].GetOperator(),
+	)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	resTx, err := testnode.SignAndBroadcastTx(encCfg, cctx.Context, accName, []sdk.Msg{msg}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+	assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the claim reward TX must succeed")
+
+	// Check if the reward amount in the account
+	balancesAfter, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+
+	assert.GreaterOrEqual(s.T(), balancesAfter.AmountOf(app.BondDenom).Int64(), minExpectedBalance, "Minimum balance after claiming reward")
+}
+
+func AddAccountsToGenesisState(gs map[string]json.RawMessage, accounts ...authtypes.GenesisAccount) (map[string]json.RawMessage, error) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	var authGenState authtypes.GenesisState
+	err := encCfg.Codec.UnmarshalJSON(gs[authtypes.ModuleName], &authGenState)
+	if err != nil {
+		return gs, err
+	}
+
+	pAccs, err := authtypes.PackAccounts(accounts)
+	if err != nil {
+		return gs, err
+	}
+
+	// set the accounts in the genesis state
+	authGenState.Accounts = append(authGenState.Accounts, pAccs...)
+	gs[authtypes.ModuleName] = encCfg.Codec.MustMarshalJSON(&authGenState)
+
+	return gs, nil
+}
+
+func AddBalancesToGenesisState(gs map[string]json.RawMessage, balances ...banktypes.Balance) (map[string]json.RawMessage, error) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	var bankGenState banktypes.GenesisState
+	err := encCfg.Codec.UnmarshalJSON(gs[banktypes.ModuleName], &bankGenState)
+	if err != nil {
+		return gs, err
+	}
+
+	bankGenState.Balances = append(bankGenState.Balances, balances...)
+	gs[banktypes.ModuleName] = encCfg.Codec.MustMarshalJSON(&bankGenState)
+
+	return gs, nil
+}
+
+func GetValidators(grpcConn *grpc.ClientConn) (stakingtypes.Validators, error) {
+	scli := stakingtypes.NewQueryClient(grpcConn)
+	vres, err := scli.Validators(context.Background(), &stakingtypes.QueryValidatorsRequest{})
+
+	if vres == nil {
+		return stakingtypes.Validators{}, err
+	}
```

Comment:
[question] does this intentionally not check if `err != nil`? When I see 

```go
foo, err := someFunc()
```

I'm conditioned to expect

```go
foo, err := someFunc()
if err != nil {
		return stakingtypes.Validators{}, err
}

Reply:
I was not sure if the `Validators` method will return a non nil `err` if the `vres` is nil, so to avoid nil reference panic on `vres.Validators` if in any case `vres` is returned nil and err is also nil due to no validators found, I check it as well and return the `err` no matter what it is.
Do you think we should have an extra `if err != nil {...`  to have a better readability?